### PR TITLE
Trunk

### PR DESCRIPTION
**Pull request by chirag-jog**
1.Add support for opsource Network APIs.
 Support added to 
- create/delete Network
- create/delete/list ACL
- create/delete/list NAT
1. Fixed a bug in error parsing.

### CODE REVIEW

File: libcloud/compute/drivers/opsource.py [hunk 1]
```diff
@@ -222,6 +222,44 @@ class OpsourceNetwork(object):
...
+    def __init__(self, id, name, status, position, action, protocol, 
+                        _type, port1, port2=None):
+        self.id = str(id)
+        self.name = name
+        self.status = status
+        self.position = position
+        self.action = action
+        self.protocol = protocol
+        self.type = _type
+        self.port1 = port1
```
Comment (Kami) [inline libcloud/compute/drivers/opsource.py:266]: Hm, one port variable outbound one inbound?

If this is the case, you should probably rename them :)

File: libcloud/compute/drivers/opsource.py [hunk 2]
```diff
@@ -380,6 +452,52 @@ class OpsourceNodeDriver(NodeDriver):
...
+        """List NAT rules for a particular network
+        """
+        return self._to_natrules(self.connection.request_with_orgId('network/%s/natrule' %network.id).object)
+
+    def ex_list_acl_rules(self, network):
+        """List ACL rules for a particular network
+        """
+        return self._to_aclrules(self.connection.request_with_orgId('network/%s/aclrule' %network.id).object)
+
+    def ex_create_nat_rule(self, network, private_ip):
```
Comment (Kami) [inline libcloud/compute/drivers/opsource.py:571]: Can you please add docstrings for the variables?

For example: 

`@type network: C{OpsourceNetwork}
@param network: Network to create a NAT rule in.
...
`

**Comment by Kami** [comment]
Can you please also add tests for all the newly added functionally?

**Comment by Kami** [comment]
Any progress on the tests or do you need help?

**Comment by chirag-jog** [comment]
Hello, 
I have been tied up with other things. Will complete the tests in the next few days. 
Thanks!

-----Original Message-----

**Pull request closed without merging**